### PR TITLE
Implement Storage

### PR DESCRIPTION
**Pull request by rocketninja7**
#119 
Fix some tests too

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonModulePlanStorage.java [hunk 1]
```diff
@@ -1,80 +1,79 @@
...
 import java.util.Optional;
 import java.util.logging.Logger;
 
-import seedu.address.commons.core.LogsCenter;
-import seedu.address.commons.exceptions.DataLoadingException;
-import seedu.address.commons.exceptions.IllegalValueException;
-import seedu.address.commons.util.FileUtil;
-import seedu.address.commons.util.JsonUtil;
-import seedu.address.model.ReadOnlyAddressBook;
+import static java.util.Objects.requireNonNull;
```
Comment (marquestye) [inline src/main/java/seedu/address/storage/JsonModulePlanStorage.java:16]: Seems to be the incorrect import ordering. The static imports should be followed by the standard java package, before the rest of the imports.

Perhaps you could run checkstylemain on gradle to verify that you have the correct import orderings.

File: src/main/java/seedu/address/storage/JsonModulePlanStorage.java [hunk 2]
```diff
@@ -1,80 +1,79 @@
...
 
-/**
- * A class to access AddressBook data stored as a json file on the hard disk.
- */
-public class JsonAddressBookStorage implements AddressBookStorage {
+public class JsonModulePlanStorage implements ModulePlanStorage {
```
Comment (marquestye) [inline src/main/java/seedu/address/storage/JsonModulePlanStorage.java:17]: Missing some javadocs.

File: src/main/java/seedu/address/storage/JsonSerializableModulePlan.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import javafx.collections.ObservableList;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.module.Module;
+import seedu.address.model.moduleplan.ModulePlan;
+import seedu.address.model.moduleplan.ModulePlanSemester;
+import seedu.address.model.moduleplan.ReadOnlyModulePlan;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
```
Comment (marquestye) [inline src/main/java/seedu/address/storage/JsonSerializableModulePlan.java:15]: Similar issue with the import ordering as above.

File: src/main/java/seedu/address/storage/ModulePlanStorage.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package seedu.address.storage;
+
+import seedu.address.commons.exceptions.DataLoadingException;
+import seedu.address.model.moduleplan.ReadOnlyModulePlan;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
```
Comment (marquestye) [inline src/main/java/seedu/address/storage/ModulePlanStorage.java:8]: Import ordering.

File: src/test/data/JsonAddressBookStorageTest/invalidAndValidModuleModulePlan.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+  "persons": [ {
```
Comment (marquestye) [inline src/test/data/JsonAddressBookStorageTest/invalidAndValidModuleModulePlan.json:2]: Should this be "modules" instead?

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 6]
```diff
@@ -10,58 +11,58 @@
 import org.junit.jupiter.api.io.TempDir;
 
 import seedu.address.commons.core.GuiSettings;
-import seedu.address.model.AddressBook;
-import seedu.address.model.ReadOnlyAddressBook;
 import seedu.address.model.UserPrefs;
+import seedu.address.model.moduleplan.ModulePlan;
+import seedu.address.model.moduleplan.ReadOnlyModulePlan;
```
Comment (marquestye) [inline src/test/java/seedu/address/storage/StorageManagerTest.java:16]: Import ordering.

**Review by marquestye — CHANGES_REQUESTED** [review]

**Comment by rocketninja7** [comment]
Import ordering fixed

**Review by marquestye — APPROVED** [review]
LGTM

**Pull request merged**